internal/command/user_grant.go

```diff
@@ -144,27 +177,33 @@ func (c *Commands) removeRoleFromUserGrant(ctx context.Context, userGrantID stri
 	return usergrant.NewUserGrantChangedEvent(ctx, userGrantAgg, existingUserGrant.UserID, existingUserGrant.RoleKeys), nil
 }
 
-func (c *Commands) DeactivateUserGrant(ctx context.Context, grantID, resourceOwner string) (objectDetails *domain.ObjectDetails, err error) {
-	if grantID == "" || resourceOwner == "" {
+func (c *Commands) DeactivateUserGrant(ctx context.Context, grantID string, resourceOwner *string, ignoreNotActive bool, check UserGrantPermissionCheck) (objectDetails *domain.ObjectDetails, err error) {
+	if grantID == "" {
 		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-N2OhG", "Errors.UserGrant.IDMissing")
 	}
 
-	existingUserGrant, err := c.userGrantWriteModelByID(ctx, grantID, resourceOwner)
+	existingUserGrant, err := c.userGrantWriteModelByID(ctx, grantID, gu.Value(resourceOwner))
 	if err != nil {
 		return nil, err
 	}
 	if existingUserGrant.State == domain.UserGrantStateUnspecified || existingUserGrant.State == domain.UserGrantStateRemoved {
 		return nil, zerrors.ThrowNotFound(nil, "COMMAND-3M9sd", "Errors.UserGrant.NotFound")
 	}
 	if existingUserGrant.State != domain.UserGrantStateActive {
+		if ignoreNotActive {
```

Comment:
Why is there now a separate handling? IMO if the usergrant is already active it can be generally be handled as there is no change necessary.

Reply:
This is about deactivating an already inactive authorization.
AFAIU this should not throw an error anymore but we don't want to break this behavior in the deprecated method.
@livio-a am I right?

Reply:
The decision was to change this behavior, as well as the "no changes" error for example, so I would just clean it up, should not make a difference for the deprecated API endpoints as it is just one error less.

Reply:
Not returning the error is breaking behavior. Why should we do that if we can easily avoid it?